### PR TITLE
Created color cycle effect

### PR DESCRIPTION
**Pull request by tangles-0**
Hi there,

I've added a color cycle effect to the app. To make this function smoothly as intended, I've bumped the default transition speed setting in default.ini from 0.00 seconds to 0.07. This seemed to have no noticeable impact on any other effects or settings - although notably did have a positive impact on use of the Music and Dominant Screen effects.

A setting to adjust the speed of the color cycle effect is desirable but a little beyond my understanding of the code used for the UI.

Cheers
Energiz3r

### CODE REVIEW

File: lifx_control_panel/frames.py [hunk 1]
```diff
@@ -584,7 +599,7 @@ def get_color_from_palette(self):
         if color:
             # RGBtoHBSK sometimes returns >65535, so we have to truncate
             hsbk = [min(c, 65535) for c in lifxlan.RGBtoHSBK(color, self.hsbk[3].get())]
-            self.set_color(hsbk)
+            self.set_color(hsbk);
```
Comment (samclane) [inline lifx_control_panel/frames.py:602]: Was there a reason for a semicolon here?

File: lifx_control_panel/utilities/color_thread.py [hunk 2]
```diff
@@ -64,6 +70,20 @@ def normalize_rectangles(rects: List[Tuple[int, int, int, int]]):
         for left, top, right, bottom in rects
     ]
 
+cycleColor = (255, 0, 0)
```
Comment (samclane) [inline lifx_control_panel/utilities/color_thread.py:73]: Should find a way to remove global values from this function.

Comment (samclane) [inline lifx_control_panel/utilities/color_thread.py:73]: Should find a way to remove global values from this function.

File: requirements.txt [hunk 3]
```diff
@@ -2,8 +2,10 @@ keyboard
...
 Pillow
-git+git://github.com/samclane/lifxlan@master#egg=lifxlan
+git+https://github.com/samclane/lifxlan@master#egg=lifxlan
 numexpr
 numpy
 mss
-pystray
\ No newline at end of file
+pystray
+pypiwin32
```
Comment (samclane) [inline requirements.txt:10]: I've read that `pywin32` is a newer version of `pypiwin32`. Is `pyaudio` only available on the latter?

Comment (samclane) [inline requirements.txt:10]: I've read that `pywin32` is a newer version of `pypiwin32`. Is `pyaudio` only available on the latter?

File: lifx_control_panel/frames.py (hunk 1 above)
Comment (Energiz3r) [reply lifx_control_panel/frames.py]: only my habits as a web developer 👀

File: lifx_control_panel/utilities/color_thread.py (hunk 2 above)
Comment (Energiz3r) [reply lifx_control_panel/utilities/color_thread.py:73]: I'm still pretty fresh to python overall - I don't know another method to store a value between calls of the function. if you can suggest some code I'd be happy to test and add it to the PR

File: requirements.txt (hunk 3 above)
Comment (Energiz3r) [reply requirements.txt]: You're correct - have fixed

**Comment by Energiz3r** [comment]
I'm not sure why Codecov reports such a large decrease in coverage of unit test but surely it's not 55%... regardless it might be wise to resolve your question regarding the global variables before writing a unit test for the color_cycle function

**Comment by samclane** [comment]
Sorry work has been hectic.

I see that Pyinstaller is failing the smoke. Speaking of work, I've just dealt with the same issue. 

I think the change that doesn't allow for `spec` files with `--onefile` parameter is a recent change. I'll find a workaround- the `bat` file was a hacky/platform-dependent choice.

**Comment by samclane** [comment]
@Energiz3r I made some PRs to your Fork that address the issues.

I'm still not sure why the CodeCov dropped; even on my recent push to `master` where I just pinned the Pyinstaller version, the coverage still plummeted to 3%. I'll ignore it for now.

**Pull request merged**